### PR TITLE
NEWS: add a bunch of stuff for the 232 release

### PR DESCRIPTION
**Pull request by keszybz**
This does not include the description of the mixed v1/v2 mode, but
everything important apart from that should be covered.

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
 
 CHANGES WITH 232 in spe
 
+        * The new RemoveIPC= option can be used to remove IPC objects owned by
+          the user or group of a service when that service exits.
+
+        * Support for dynamically creating users for the lifetime of a service
+          has been added. If DynamicUser=yes is specified, a user and group IDs
```
Comment (martinpitt) [inline NEWS]: "ID" (singular), or drop the "a" (plural), but I think the former sounds nicer.

File: NEWS [hunk 2]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+          will be allocated from the range 61184..65519 for the lifetime of the
+          service. They can be resolved using the new nss-systemd.so NSS
+          module. The module must be enabled in /etc/nsswitch.conf. Services
+          started in this way have PrivateTmp= and RemoveIPC= enabled, so that
+          any resources allocated by the service will be cleaned up when the
+          service exits.
+
+          nss-systemd module also always resolves root and nobody, making it
```
Comment (martinpitt) [inline NEWS]: "The" nss-...

File: NEWS [hunk 3]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+          possible to have no /etc/passwd or /etc/group files in minimal
+          container systems.
```
Comment (martinpitt) [inline NEWS:19]: Hmm, most distros will require additional static GIDs. http://refspecs.linuxbase.org/LSB_4.1.0/LSB-Core-generic/LSB-Core-generic/usernames.html at least requires `bin` and `daemon` (although I'm not aware of anything actually using those), but e. g. `adm` is even being used by the journal. So I wonder if it is still premature to advertise this yet?

File: NEWS [hunk 4]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+          PrivateUsers= option. Only root, nobody, and the uid/gid under which
+          the service is running are mapped. All other users are mapped to
+          nobody.
+
+        * Support for the cgroup namespace has been added to systemd-nspawn. If
+          supported by kernel, the container system started by systemd-nspawn
+          will have its own view of the cgroup hierarchy. This new behaviour
+          can be disabled using $SYSTEMD_NSPAWN_USE_CGNS environment variable.
+
+          The new MemorySwapMax= option can be used to limit the maximum swap
```
Comment (martinpitt) [inline NEWS]: Missing "*" itemizer

File: NEWS [hunk 5]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+          usage under the unified cgroup hierarchy.
+
+          Support for the CPU controller in the unified cgroup hierarchy has
```
Comment (martinpitt) [inline NEWS]: Dito

File: NEWS [hunk 6]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+          been added, using the CPUWeight=, CPUStartupWeight=, CPUAccounting=
```
Comment (martinpitt) [inline NEWS]: "using" sounds a bit strange -- s/using/via/ or s/, using/through the../ ?

File: NEWS [hunk 7]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+          options. This controller requires out-of-tree patches for the kernel
+          and the support is provisional.
+
+        * .automount units may now be transient.
+
+          systemd-mount is a new tool which wraps mount(8) to pull in
```
Comment (martinpitt) [inline NEWS]: missing "*" itemizer

File: NEWS [hunk 8]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+          additional dependencies through transient .mount and .automount
+          units. For example fsck is done automatically on the block device
```
Comment (martinpitt) [inline NEWS]: "For example, this automatically runs fsck on..."?

File: NEWS [hunk 9]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+          the directory is present, and the mount point was not configured
+          through other means (e.g. fstab). If /efi directory does not exist,
+          /boot will be used as before. This makes it easier to automatically
+          mount the EFI partition on systems where /boot is used for something
+          else.
+
+        * disk/by-id symlinks are now created for NVMe drives.
+
+        * A set of new user session targets have been added to support running
+          users' graphical sessions under the systemd --user instance:
```
Comment (martinpitt) [inline NEWS]: I'd drop the " users' " here.

File: NEWS [hunk 10]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+
+        * Information recored in the journal for core dumps now includes the
+          contents of /proc/mountinfo and the command line of the process at
+          the top of the process hierarchy (which is usually the init process
+          of the container).
+
+        * systemd-journal-gatewayd learned the --directory option to serve
+          files from the specified location.
+
+          journalctl --root=… can be used to peruse journal in the /var/log/
```
Comment (martinpitt) [inline NEWS]: - itemizer

peruse _the_ journal

File: NEWS [hunk 11]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+          directories inside of a container tree. This is similar to the
+          existing --machine= option, but does not require the container to be
+          active.
+
+        * The hardware database has been extended to support
+          ID_INPUT_TRACKBALL, used in addition to ID_INPUT_MOUSE to identify
+          trackball devices.
+
+          MOUSE_WHEEL_CLICK_ANGLE_HORIZONTAL hwdb property has been added to
```
Comment (martinpitt) [inline NEWS:89]: "*" itemizer

File: NEWS [hunk 12]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+        * A new journal output mode "short-full" has been added which uses
+          timestamps with abbreviated English day names and adds a timezone
+          suffix. Those timestamps include more information and can be parsed
+          by journalctl.
+
+        * /etc/resolv.conf will be bind-mounted into containers started by
+          systemd-nspawn, if possible, so any changes to resolv.conf contents
+          are automatically propagated to the container.
+
+        * The number of instances for socket-activated services originiating
```
Comment (martinpitt) [inline NEWS]: "originating" (extra 'i')

File: NEWS [hunk 13]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+          extending the existing setting of MaxConnections.
+
+        * UDP Segmentation Offload, TCP Segmentation Offload, Generic
+          Segmentation Offload, Generic Receive Offload, Large Receive Offload
+          can be enabled and disabled using the new UDPSegmentationOffload=,
+          TCPSegmentationOffload=, GenericSegmentationOffload=,
+          GenericReceiveOffload=, LargeReceiveOffload= options in the
+          [Link] section of .link files.
+
+          Spanning Tree Protocol enablement, Priority, Aging Time, and the
```
Comment (martinpitt) [inline NEWS:116]: missing itemizer

File: NEWS [hunk 14]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+          Default Port VLAN ID can be configured for bridge devices using the
+          new STP=, Priority=, AgeingTimeSec=, and DefaultPVID= settings in the
+          [Bridge] section of .netdev files.
+
+          Address Resolution Protocol can be disabled on links managed by
```
Comment (martinpitt) [inline NEWS:121]: itemizer

File: NEWS [hunk 15]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+          systemd-networkd using the ARP=no setting in the [Link] section of
+          .network files.
+
+        * $SERVICE_RESULT, $EXIT_CODE, $EXIT_STATUS are set for ExecStop= and
+          ExecStopPost= commands.
+
+        * configure learned the --with-support-url= option to specify the
+          distribution's bugtracker.
```
Comment (martinpitt) [inline NEWS]: This already got released in 231.

File: NEWS [hunk 16]
```diff
@@ -2,11 +2,144 @@ systemd System and Service Manager
...
+
         * Journald's SplitMode=login setting has been deprecated. It has been
           removed from documentation, and it's use is discouraged. In a future
           release it will be completely removed, and made equivalent to current
           default of SplitMode=uid.
 
+        * The --share-system option has been replaced with an (undocumented)
```
Comment (martinpitt) [inline NEWS]: I suppose this means "--share-system systemd-nspawn option"?

**Comment by martinpitt** [comment]
Thanks for the writeup! Found a couple of issues.

File: NEWS (hunk 3 above)
Comment (keszybz) [reply NEWS:19]: This only applies to "minimal" containers… So no journald inside. Maybe this should be worded in some better way, I'm open to suggestions.

LSB says that both `bin` and `daemon` are legacy and should not be used in new applications. So I think we can safely ignore LSB here.

File: NEWS (hunk 15 above)
Comment (keszybz) [reply NEWS]: Oh, it was only added to the porting guide. But that's useful to advertise, so I'll move this entry down into v231 news.

**Comment by keszybz** [comment]
Thanks for the detailed review. I followed or responded to all comments, except for the missing itemizers. They were omitted in some places on purpose, to have "multi-paragraph" entries when there are a few closely related items. Nevertheless, I added them in a few places, where it made most sense, so now they are only omitted in the few places where the items are most closely related.

**Pull request merged**